test/Microsoft.AspNet.Mvc.FunctionalTests/InlineConstraintTests.cs

```diff
@@ -478,6 +478,35 @@ public class InlineConstraintTests
             Assert.Equal(HttpStatusCode.NotFound, response.StatusCode);
         }
 
+        [Theory]
+        // Testing custom inline constraint configured via MapRoute
+        [InlineData("software", HttpStatusCode.OK, "software From Software Controller Index")]
+        // Testing custom inline constraint updated in ConstraintMap
+        [InlineData("hardware", HttpStatusCode.NotFound, null)]
+        [InlineData("HARDWARE", HttpStatusCode.OK, "HARDWARE From Hardware Controller Index")]
+        // Testing custom inline constraint configured with Attribute Routing
+        [InlineData("consumable", HttpStatusCode.OK, "consumable From Consumable Controller Index")]
+        public async Task CustomInlineConstraint_Add_Update(
```

Comment:
There is no test for removing constraints from ConstraintMap. I think that is not a good candidate for functional tests.  Once a constraint is removed, and it is still used in the Web Site, it will simply return 500 error. So, I think I need to create another Web site only for that scenario. I think there should be no special case from custom inline constraints' perspective. 
But, please let me know if you think we need to add a test for that scenario too.